unyt/array.py

```diff
@@ -1661,15 +1624,8 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
                         raise InvalidUnitOperation(
                             "Quantities with units of Fahrenheit or Celsius "
                             "cannot by multiplied, divided, subtracted or "
-                            "added.")
+                            "added with data that has different units.")
                     inp1 = np.asarray(inp1)*conv
-                else:
-                    if ((u0.base_offset and u0.dimensions is temperature or
```

Comment:
Are we allowing this now?

Reply:
https://github.com/yt-project/yt/issues/1785